modules/core/shared/src/main/scala/io/circe/Printer.scala

```diff
@@ -242,12 +255,22 @@ final case class Printer(
     }
   }
 
+  @transient
+  private[this] final val stringWriter: ThreadLocal[StringBuilder] = new ThreadLocal[StringBuilder] {
+    override final def initialValue: StringBuilder = new StringBuilder()
+  }
+
   /**
    * Returns a string representation of a pretty-printed JSON value.
    */
   final def pretty(json: Json): String = {
-    val writer = new StringBuilder()
-    val folder = new PrintingFolder(writer)
+    val writer = if (reuseWriters) {
```

Comment:
Ah right. I have a strong feeling that as of this change string-base printing will probably be a best choice for the TE benchmark given we send the same message over and over again. The fact we can cache a `StringBuilder` means we don't need to resize/grow it for the next message (given it's the same message). This should translate into a lot of RPS.

I'm thinking of something like message-size-guestimation machinery would do the same trick for binary-printing.